test/unit/puppetlabs/services/jruby_pool_manager/jruby_agents_test.clj

```diff
@@ -68,7 +74,10 @@
             jruby-instance
             pool-context
             :with-jruby-retry-test
-            (is (instance? JRubyInstance jruby-instance))))
+            (is (instance? JRubyInstance jruby-instance)
+                (format
+                 "Unable to borrow instance after retries.  Num borrows: %d"
+                 @num-borrows))))
```

Comment:
One thing that I noticed in the test that hung before is that the default borrow timeout is 20 minutes, but also the jenkins job will timeout if it doesn't receive output for 20 minutes. This is a bigger thing than just this test, but maybe it would be a good idea to lower the default borrow timeout in the test config?